File: app/src/main/java/com/supercasual/fourtop/uimain/TopUsersFragment.java
package com.supercasual.fourtop.uimain;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;

import com.supercasual.fourtop.R;
import com.supercasual.fourtop.databinding.FragmentTopUsersBinding;

import org.jetbrains.annotations.NotNull;

public class TopUsersFragment extends Fragment {

    private FragmentTopUsersBinding binding;

    @Override
    public View onCreateView(@NotNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_top_users, container,
                false);
        return binding.getRoot();
    }
}

File: app/src/main/java/com/emikhalets/voteapp/view/top_users/TopUsersFragment.kt
package com.emikhalets.voteapp.view.top_users

import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import by.kirich1409.viewbindingdelegate.viewBinding
import com.emikhalets.voteapp.R
import com.emikhalets.voteapp.databinding.FragmentTopUsersBinding
import com.emikhalets.voteapp.test.createMockUsers
import com.emikhalets.voteapp.utils.ACTIVITY
import com.emikhalets.voteapp.view.adapters.UsersAdapter
import com.emikhalets.voteapp.view.base.SecondaryFragment

class TopUsersFragment : SecondaryFragment(R.layout.fragment_top_users) {

    private val binding: FragmentTopUsersBinding by viewBinding()
    private lateinit var usersAdapter: UsersAdapter
    private lateinit var llm: LinearLayoutManager

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        ACTIVITY.title = getString(R.string.top_users_title)
        initRecyclerView()
    }

    private fun initRecyclerView() {
        usersAdapter = UsersAdapter()
        llm = LinearLayoutManager(this.context)
        binding.apply {
            listUsers.layoutManager = llm
            listUsers.setHasFixedSize(true)
            listUsers.isNestedScrollingEnabled = false
            listUsers.adapter = usersAdapter
        }
        usersAdapter.updateList(createMockUsers().sortedByDescending { it.rating })
    }
}